diff --git a/Character/Look/Body.cpp.rej b/Character/Look/Body.cpp.rej
new file mode 100644
index 0000000..0178a05
--- /dev/null
+++ b/Character/Look/Body.cpp.rej
@@ -0,0 +1,10 @@
+diff a/Character/Look/Body.cpp b/Character/Look/Body.cpp	(rejected hunks)
+@@ -118,7 +118,7 @@ namespace ms
+ 		{
+ 			LOG(LOG_DEBUG, "Skin [" << skin << "] is using the default value.");
+ 
+-			name = nl::nx::String["Eqp.img"]["Eqp"]["Skin"][skin]["name"];
++			name = nl::nx::String["Eqp.img"]["Eqp"]["Skin"][skin]["name"].get_string();
+ 		}
+ 	}
+ 
