diff --git a/Character/Look/Clothing.cpp.rej b/Character/Look/Clothing.cpp.rej
new file mode 100644
index 0000000..0508781
--- /dev/null
+++ b/Character/Look/Clothing.cpp.rej
@@ -0,0 +1,10 @@
+diff a/Character/Look/Clothing.cpp b/Character/Look/Clothing.cpp	(rejected hunks)
+@@ -85,7 +85,7 @@ namespace ms
+ 		nl::node src = nl::nx::Character[category][strid + ".img"];
+ 		nl::node info = src["info"];
+ 
+-		vslot = info["vslot"];
++		vslot = std::string(info["vslot"]);
+ 
+ 		switch (int32_t standno = info["stand"])
+ 		{
