diff --git a/Character/Look/Hair.cpp.rej b/Character/Look/Hair.cpp.rej
new file mode 100644
index 0000000..4e6122f
--- /dev/null
+++ b/Character/Look/Hair.cpp.rej
@@ -0,0 +1,10 @@
+diff a/Character/Look/Hair.cpp b/Character/Look/Hair.cpp	(rejected hunks)
+@@ -98,7 +98,7 @@ namespace ms
+ 			}
+ 		}
+ 
+-		name = nl::nx::String["Eqp.img"]["Eqp"]["Hair"][std::to_string(hairid)]["name"];
++		name = nl::nx::String["Eqp.img"]["Eqp"]["Hair"][std::to_string(hairid)]["name"].get_string();
+ 
+ 		constexpr size_t NUM_COLORS = 8;
+ 
