diff --git a/Data/JobData.cpp.rej b/Data/JobData.cpp.rej
new file mode 100644
index 0000000..fb08274
--- /dev/null
+++ b/Data/JobData.cpp.rej
@@ -0,0 +1,10 @@
+diff a/Data/JobData.cpp b/Data/JobData.cpp	(rejected hunks)
+@@ -33,7 +33,7 @@ namespace ms
+ 
+ 		icon = src["info"]["icon"];
+ 
+-		name = strsrc["bookName"];
++		name = strsrc["bookName"].get_string();
+ 
+ 		for (nl::node sub : src["skill"])
+ 		{
