diff --git a/Data/ItemData.cpp.rej b/Data/ItemData.cpp.rej
new file mode 100644
index 0000000..60ff187
--- /dev/null
+++ b/Data/ItemData.cpp.rej
@@ -0,0 +1,12 @@
+diff a/Data/ItemData.cpp b/Data/ItemData.cpp	(rejected hunks)
+@@ -78,8 +78,8 @@ namespace ms
+ 			cashitem = src["cash"].get_bool();
+ 			gender = get_item_gender(itemid);
+ 
+-			name = strsrc["name"];
+-			desc = strsrc["desc"];
++			name = strsrc["name"].get_string();
++			desc = strsrc["desc"].get_string();
+ 
+ 			valid = true;
+ 		}
