diff --git a/Gameplay/MapleMap/Npc.cpp b/Gameplay/MapleMap/Npc.cpp
index b7150c6..54010c5 100644
--- a/Gameplay/MapleMap/Npc.cpp
+++ b/Gameplay/MapleMap/Npc.cpp
@@ -62,8 +62,8 @@ namespace ms
 				lines[state].push_back(strsrc[speaknode.get_string()]);
 		}
 
-		name = strsrc["name"];
-		func = strsrc["func"];
+		name = strsrc["name"].get_string();
+		func = strsrc["func"].get_string();
 
 		std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
 		std::wstring wide = converter.from_bytes(func);
