diff --git a/Gameplay/Combat/SkillAction.cpp b/Gameplay/Combat/SkillAction.cpp
index 2988c89..1be8778 100644
--- a/Gameplay/Combat/SkillAction.cpp
+++ b/Gameplay/Combat/SkillAction.cpp
@@ -44,7 +44,7 @@ namespace ms
 
 	SingleAction::SingleAction(nl::node src)
 	{
-		action = src["action"]["0"];
+		action = src["action"]["0"].get_string();
 	}
 
 	void SingleAction::apply(Char& target, Attack::Type) const
@@ -54,8 +54,8 @@ namespace ms
 
 	TwoHandedAction::TwoHandedAction(nl::node src)
 	{
-		actions[false] = src["action"]["0"];
-		actions[true] = src["action"]["1"];
+		actions[false] = src["action"]["0"].get_string();
+		actions[true] = src["action"]["1"].get_string();
 	}
 
 	void TwoHandedAction::apply(Char& target, Attack::Type) const
@@ -71,7 +71,7 @@ namespace ms
 		for (auto sub : src["level"])
 		{
 			int32_t level = string_conversion::or_zero<int32_t>(sub.name());
-			actions[level] = sub["action"];
+			actions[level] = sub["action"].get_string();
 		}
 
 		skillid = id;
