diff --git a/includes/glew-2.1.0/include/GL/glew.h b/includes/glew-2.1.0/include/GL/glew.h
index b5b6987..1138ee4 100644
--- a/includes/glew-2.1.0/include/GL/glew.h
+++ b/includes/glew-2.1.0/include/GL/glew.h
@@ -81,6 +81,27 @@
 #define __glew_h__
 #define __GLEW_H__
 
+#ifdef MS_PLATFORM_WASM
+// MS WASM: Stubbing GLEW for WASM/Emscripten which uses GLES3 directly
+#include <GLES3/gl3.h>
+#define GLEW_OK 0
+#define GLEW_NO_ERROR 0
+#define GLEW_VERSION 1
+#define GLEW_VERSION_MAJOR 2
+#define GLEW_VERSION_MINOR 1
+#define GLEW_VERSION_MICRO 0
+
+static int glewInit() { return 0; }
+static int glewIsSupported(const char* name) { return 0; }
+#define glewGetErrorString(x) (const GLubyte*)"(WASM: No GLEW Error)"
+#define glewGetString(x) (const GLubyte*)"(WASM: No GLEW String)"
+
+static int glewExperimental = 0;
+
+typedef GLuint GLhandleARB; // Compatibility
+
+#else
+
 #if defined(__gl_h_) || defined(__GL_H__) || defined(_GL_H) || defined(__X_GL_H)
 #error gl.h included before glew.h
 #endif
@@ -23684,3 +23705,5 @@ GLEWAPI const GLubyte * GLEWAPIENTRY glewGetString (GLenum name);
 /* #undef GLEWAPI */
 
 #endif /* __glew_h__ */
+
+#endif
