diff --git a/includes/NoLifeNx/nlnx/node.hpp b/includes/NoLifeNx/nlnx/node.hpp
index ac8d107..4d0a548 100644
--- a/includes/NoLifeNx/nlnx/node.hpp
+++ b/includes/NoLifeNx/nlnx/node.hpp
@@ -132,7 +132,11 @@ namespace nl {
         bitmap to_bitmap() const;
         audio to_audio() const;
         //Internal variables
+#ifdef MS_PLATFORM_WASM
+        uint32_t m_offset = 0;
+#else
         data const * m_data = nullptr;
+#endif
         _file_data const * m_file = nullptr;
         friend file;
     };
