diff --git a/includes/NoLifeNx/nlnx/nx.hpp b/includes/NoLifeNx/nlnx/nx.hpp
new file mode 100644
index 0000000..be7f6bd
--- /dev/null
+++ b/includes/NoLifeNx/nlnx/nx.hpp
@@ -0,0 +1,33 @@
+//////////////////////////////////////////////////////////////////////////////
+// NoLifeNx - Part of the NoLifeStory project                               //
+// Copyright © 2013 Peter Atashian                                          //
+//                                                                          //
+// This program is free software: you can redistribute it and/or modify     //
+// it under the terms of the GNU Affero General Public License as           //
+// published by the Free Software Foundation, either version 3 of the       //
+// License, or (at your option) any later version.                          //
+//                                                                          //
+// This program is distributed in the hope that it will be useful,          //
+// but WITHOUT ANY WARRANTY; without even the implied warranty of           //
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            //
+// GNU Affero General Public License for more details.                      //
+//                                                                          //
+// You should have received a copy of the GNU Affero General Public License //
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.    //
+//////////////////////////////////////////////////////////////////////////////
+#pragma once
+#include "nxfwd.hpp"
+
+namespace nl
+{
+	namespace nx
+	{
+		// Pre-defined nodes to access standard MapleStory style data
+		// Make sure you called load_all first
+		extern node Base, Character, Effect, Etc, Item, Map, Map001, Map002, Map2, Mob, Mob001, Mob002, Mob2, Morph, Npc, Quest, Reactor, Skill, Skill001, Skill002, Skill003, Sound, Sound001, Sound002, Sound2, String, TamingMob, UI;
+
+		// Loads the pre-defined nodes from a standard setup of nx files for MapleStory
+		// Only call this function once
+		void load_all();
+	}
+}
\ No newline at end of file
