diff --git a/includes/NoLifeNx/nlnx/bitmap.cpp b/includes/NoLifeNx/nlnx/bitmap.cpp
index a2850db..e7de766 100644
--- a/includes/NoLifeNx/nlnx/bitmap.cpp
+++ b/includes/NoLifeNx/nlnx/bitmap.cpp
@@ -17,42 +17,64 @@
 //////////////////////////////////////////////////////////////////////////////
 
 #include "bitmap.hpp"
+#include "file_impl.hpp"
 #include <lz4.h>
+#include <unistd.h>
 #include <vector>
 
-namespace nl {
-    bitmap::bitmap(void const * d, uint16_t w, uint16_t h) :
-        m_data(d), m_width(w), m_height(h) {}
-    bool bitmap::operator<(bitmap const & o) const {
-        return m_data < o.m_data;
-    }
-    bool bitmap::operator==(bitmap const & o) const {
-        return m_data == o.m_data;
-    }
-    bitmap::operator bool() const {
-        return m_data ? true : false;
-    }
-    std::vector<char> bitmap_buf;
-    void const * bitmap::data() const {
-        if (!m_data)
-            return nullptr;
-        auto const l = length();
-        if (l + 0x20 > bitmap_buf.size())
-            bitmap_buf.resize(l + 0x20);
-        ::LZ4_decompress_fast(4 + reinterpret_cast<char const *>(m_data),
-            bitmap_buf.data(), static_cast<int>(l));
-        return bitmap_buf.data();
-    }
-    uint16_t bitmap::width() const {
-        return m_width;
-    }
-    uint16_t bitmap::height() const {
-        return m_height;
-    }
-    uint32_t bitmap::length() const {
-        return 4u * m_width * m_height;
-    }
-    size_t bitmap::id() const {
-        return reinterpret_cast<size_t>(m_data);
-    }
-}
+namespace nl
+{
+	bitmap::bitmap(void* f, uint64_t o, uint16_t w, uint16_t h) : m_file(f), m_offset(o), m_width(w), m_height(h)
+	{
+	}
+	bool bitmap::operator<(bitmap const& o) const
+	{
+		return m_offset < o.m_offset;
+	}
+	bool bitmap::operator==(bitmap const& o) const
+	{
+		return m_offset == o.m_offset;
+	}
+	bitmap::operator bool() const
+	{
+		return m_file != nullptr;
+	}
+	std::vector<char> bitmap_buf;
+	void const* bitmap::data() const
+	{
+		if (!m_file)
+			return nullptr;
+		_file_data* fd = static_cast<_file_data*>(m_file);
+
+		auto const l = length();
+		if (l + 0x20 > bitmap_buf.size())
+			bitmap_buf.resize(l + 0x20);
+
+		uint64_t data_offset = m_offset + 4;
+
+		uint32_t read_len = l;
+		std::vector<char> src_buf(read_len);
+
+		if (!io_read(fd, src_buf.data(), read_len, data_offset))
+			return nullptr;
+
+		::LZ4_decompress_fast(src_buf.data(), bitmap_buf.data(), static_cast<int>(l));
+		return bitmap_buf.data();
+	}
+	uint16_t bitmap::width() const
+	{
+		return m_width;
+	}
+	uint16_t bitmap::height() const
+	{
+		return m_height;
+	}
+	uint32_t bitmap::length() const
+	{
+		return 4u * m_width * m_height;
+	}
+	size_t bitmap::id() const
+	{
+		return static_cast<size_t>(m_offset);
+	}
+} // namespace nl
