diff --git a/IO/UITypes/UIChatBar.cpp b/IO/UITypes/UIChatBar.cpp
index ab7cad1..9a60d3e 100644
--- a/IO/UITypes/UIChatBar.cpp
+++ b/IO/UITypes/UIChatBar.cpp
@@ -32,10 +32,12 @@ namespace jrc
         position = pos;
         dimension = { 500, 60 };
         chatopen = true;
+        chattarget = CHT_ALL;
         chatrows = 4;
         rowpos = 0;
         rowmax = -1;
         lastpos = 0;
+        dragchattop = false;
 
         nl::node mainbar = nl::nx::ui["StatusBar2.img"]["mainBar"];
 
@@ -138,7 +140,10 @@ namespace jrc
 
             slider.draw({ position.x(), getchattop() + 5 });
 
-            chattargets[chattarget].draw(position + Point<int16_t>(0, 2));
+            if (chattarget >= CHT_ALL && chattarget < NUM_TARGETS)
+            {
+                chattargets[chattarget].draw(position + Point<int16_t>(0, 2));
+            }
             chatcover.draw(position);
             chatfield.draw(position);
         }
@@ -155,6 +160,11 @@ namespace jrc
         chatfield.update(position);
     }
 
+    void UIChatbar::set_position(Point<int16_t> pos)
+    {
+        position = pos;
+    }
+
     Button::State UIChatbar::button_pressed(uint16_t id)
     {
         switch (id)
