diff --git a/IO/UITypes/UICharSelect.cpp b/IO/UITypes/UICharSelect.cpp
index 0fb952f..633d5d1 100644
--- a/IO/UITypes/UICharSelect.cpp
+++ b/IO/UITypes/UICharSelect.cpp
@@ -35,7 +35,10 @@
 #include "../../Net/Packets/SelectCharPackets.h"
 
 #define NOMINMAX
+
+#ifndef MS_PLATFORM_WASM
 #include <windows.h>
+#endif
 
 #ifdef USE_NX
 #include <nlnx/nx.hpp>
@@ -43,8 +46,7 @@
 
 namespace ms
 {
-	UICharSelect::UICharSelect(std::vector<CharEntry> characters, int8_t characters_count, int32_t slots, int8_t require_pic) : UIElement(Point<int16_t>(0, 0), Point<int16_t>(800, 600)),
-		characters(characters), characters_count(characters_count), slots(slots), require_pic(require_pic), tab_index(0), tab_active(false), tab_move(false), charslot_y(0), use_timestamp(false), burning_character(true), show_pic_btns(false)
+	UICharSelect::UICharSelect(std::vector<CharEntry> characters, int8_t characters_count, int32_t slots, int8_t require_pic) : UIElement(Point<int16_t>(0, 0), Point<int16_t>(800, 600)), characters(characters), characters_count(characters_count), slots(slots), require_pic(require_pic), tab_index(0), tab_active(false), tab_move(false), charslot_y(0), use_timestamp(false), burning_character(true), show_pic_btns(false)
 	{
 		std::string version_text = Configuration::get().get_version();
 		version = Text(Text::Font::A12B, Text::Alignment::LEFT, Color::Name::LEMONGRASS, "Ver. " + version_text);
@@ -790,7 +792,11 @@ namespace ms
 			{
 				std::string url = Configuration::get().get_resetpic();
 
+#ifdef MS_PLATFORM_WASM
+				LOG(LOG_ERROR, "Buttons::BtResetPIC not implemented in WASM");
+#else
 				ShellExecuteA(NULL, "open", url.c_str(), NULL, NULL, SW_SHOWNORMAL);
+#endif
 				break;
 			}
 			case Buttons::BtCharacter:
