diff --git a/IO/UITypes/UISoftKey.h b/IO/UITypes/UISoftKey.h
index 4da41e6..d27a653 100644
--- a/IO/UITypes/UISoftKey.h
+++ b/IO/UITypes/UISoftKey.h
@@ -64,7 +64,7 @@ namespace jrc
             BT_OK
         };
 
-        static constexpr size_t MIN_SIZE = 6;
+        static constexpr size_t MIN_SIZE = 4;
         static constexpr size_t MAX_SIZE = 12;
         static constexpr uint8_t NUM_KEYS = 10;
 
@@ -73,4 +73,3 @@ namespace jrc
         Randomizer random;
     };
 }
-
