diff --git a/Net/Packets/SelectCharPackets.h b/Net/Packets/SelectCharPackets.h
index c303ab0..9fc2b08 100644
--- a/Net/Packets/SelectCharPackets.h
+++ b/Net/Packets/SelectCharPackets.h
@@ -21,7 +21,7 @@
 namespace jrc
 {
     constexpr auto MACS = "68-5D-43-F8-B8-6C, 7A-79-19-8B-31-3F";
-    constexpr auto HWID = "685D43F8_B86C7A79";
+    constexpr auto HWID = "685D43F8B86C_B86C7A79";
 
     // Tell the server which character was picked.
     // Opcode: SELECT_CHAR(19)
