diff --git a/Net/Session.cpp b/Net/Session.cpp
index 03a1df7..199a70e 100644
--- a/Net/Session.cpp
+++ b/Net/Session.cpp
@@ -18,10 +18,13 @@
 #include "Session.h"
 
 #include "../Configuration.h"
+#include "MapleStory.h"
+#include "../Util/Misc.h"
 
 namespace ms
 {
-	Session::Session()
+	// TODO: Move socket info to Configuration
+	Session::Session() : socket() // WebSocket proxy address
 	{
 		connected = false;
 		length = 0;
@@ -36,14 +39,26 @@ namespace ms
 
 	bool Session::init(const char* host, const char* port)
 	{
+		std::string final_host = host;
+		if (final_host == "0.0.0.0")
+		{
+			final_host = getBrowserHostname();
+			LOG(LOG_DEBUG, "Host is 0.0.0.0, resolving to: " << final_host);
+		}
+
+		LOG(LOG_INFO, "Connecting to " << final_host << ":" << port);
 		// Connect to the server
-		connected = socket.open(host, port);
+		connected = socket.open(final_host.c_str(), port);
 
 		if (connected)
 		{
 			// Read keys necessary for communicating with the server
 			cryptography = { socket.get_buffer() };
 		}
+		else
+		{
+			LOG(LOG_ERROR, "Failed to connect");
+		}
 
 		return connected;
 	}
@@ -61,6 +76,7 @@ namespace ms
 
 	void Session::reconnect(const char* address, const char* port)
 	{
+		LOG(LOG_INFO, "Reconnecting to " << address << ":" << port);
 		// Close the current connection and open a new one
 		bool success = socket.close();
 
