diff --git a/Util/NxFiles.cpp b/Util/NxFiles.cpp
index e5544d1..d763195 100644
--- a/Util/NxFiles.cpp
+++ b/Util/NxFiles.cpp
@@ -24,14 +24,28 @@
 
 #include <fstream>
 
+#ifdef MS_PLATFORM_WASM
+#include "../LazyFS/LazyFS.h"
+#endif
+
 #include <unistd.h>
 #include <cstdio>
 
-
 namespace jrc
 {
     Error NxFiles::init()
     {
+#ifdef MS_PLATFORM_WASM
+        LazyFS::Initialize();
+        for (auto filename : NxFiles::filenames)
+        {
+            std::string url = std::string("/assets/") + filename;
+            if (!LazyFS::RegisterFile(filename, url))
+            {
+                return { Error::MISSING_FILE, filename };
+            }
+        }
+#else
         for (auto filename : NxFiles::filenames)
         {
             if (!std::ifstream{ filename }.good())
@@ -39,6 +53,7 @@ namespace jrc
                 return { Error::MISSING_FILE, filename };
             }
         }
+#endif
 
         try
         {
@@ -53,14 +68,14 @@ namespace jrc
 
         constexpr const char* POSTCHAOS_BITMAP =
             "Login.img/WorldSelect/BtChannel/layer:bg";
-        auto postChaosBitmapType =
-            nl::nx::ui.resolve(POSTCHAOS_BITMAP).data_type();
-
-        if (postChaosBitmapType != nl::node::type::bitmap)
+        // Attempt to resolve it to log if this "version" identifier exists
+        auto ui_node = nl::nx::ui.resolve(POSTCHAOS_BITMAP);
+        
+        if (!ui_node || ui_node.data_type() != nl::node::type::bitmap)
         {
-            return Error::WRONG_UI_FILE;
+            Console::get().print("[init] UI.nx is missing the post-Chaos 'Login.img/WorldSelect/BtChannel/layer:bg' node.");
+            Console::get().print("[init] Client expected post-Chaos UI.nx version, but continuing anyway.");
         }
-
         return Error::NONE;
     }
 }
