diff --git a/sql/db_shopupdate.sql b/sql/db_shopupdate.sql
index fbf4e67d..4aedfa23 100644
--- a/sql/db_shopupdate.sql
+++ b/sql/db_shopupdate.sql
@@ -1,7 +1,7 @@
 #THIS SQL IS OPTIONAL, TO BE USED AFTER 'db_drops.sql'
 #THIS REQUIRES PROVIDED WZ FILES
 
-USE `heavenms`;
+USE `maplestory`;
 
 # Scroll shop at Asia, chair shop at Kino Konoko, potion shop at T-1337
 INSERT INTO `shops` (`shopid`,`npcid`) VALUES
